target/linux/generic/pending-6.12/720-06-net-phy-realtek-support-interrupt-of-RTL8221B.patch

```diff
@@ -73,30 +21,12 @@ Signed-off-by: Jianhui Zhao <zhaojh329@gmail.com>
  		.soft_reset     = genphy_soft_reset,
  		.probe		= rtl822x_probe,
  		.get_features   = rtl822x_get_features,
-@@ -1797,6 +1844,8 @@ static struct phy_driver realtek_drvs[]
+@@ -2077,6 +2079,8 @@ static struct phy_driver realtek_drvs[]
  	}, {
- 		.match_phy_device = rtl8221b_vb_cg_c45_match_phy_device,
- 		.name           = "RTL8221B-VB-CG 2.5Gbps PHY (C45)",
-+		.config_intr	= rtl8221b_config_intr,
-+		.handle_interrupt = rtl8221b_handle_interrupt,
- 		.soft_reset     = rtl822x_c45_soft_reset,
- 		.probe		= rtl822x_probe,
- 		.config_init    = rtl822xb_config_init,
-@@ -1809,6 +1858,8 @@ static struct phy_driver realtek_drvs[]
- 	}, {
- 		.match_phy_device = rtl8221b_vn_cg_c22_match_phy_device,
+ 		.match_phy_device = rtl8221b_vm_cg_c22_match_phy_device,
  		.name           = "RTL8221B-VM-CG 2.5Gbps PHY (C22)",
 +		.config_intr	= rtl8221b_config_intr,
 +		.handle_interrupt = rtl8221b_handle_interrupt,
  		.soft_reset     = genphy_soft_reset,
  		.probe		= rtl822x_probe,
  		.get_features   = rtl822x_get_features,
```

Comment:
In the upstream version of this patch you said it is not possible to access these registers in C22 mode, see: https://lore.kernel.org/all/20251102152644.1676482-1-olek2@wp.pl/
Why do you keep this patch?

Reply:
By the way, recently I found that the C22 variant can read C45 registers via page operations. I wrote a [proof of concept ](https://github.com/abajk/linux/commit/59cb369ae903c54f566d05c8bbd760fb1cdfeef0)that fixes interrupts and temperature reading. I'm still thinking how to do it properly.

Reply:
Do we have devices which make use the interrupt handler on the PHY in C22 mode?

It would be nice if you could update the patch description. It should contain why we add this for C22 too and why it is save.